### PR TITLE
chore(examples): rspack with lingui swc plugin

### LINKED ISSUE #1896: rspack now supports wasm plugins, lingui integration example could be improved
https://www.rspack.dev/guide/builtin-swc-loader.html#optionsjscexperimentalplugins

Babel could be dropped, and swc-loader could b directly used with lingui swc plugin. Should result in a massive performance improvement.

### PR DESCRIPTION
**Pull request by aseerkt**
# Description
Replace babel-loader with builtin swc-loader of rspack
[Screencast from 19-04-24 08:07:52 PM IST.webm](https://github.com/lingui/js-lingui/assets/39845171/f94cd41f-1cc6-4a2b-b0b7-8e8423f0d8bb)

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Examples update

Fixes #1896

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)

### CODE REVIEW

File: examples/rspack/rspack.config.js [hunk 1]
```diff
@@ -1,40 +1,53 @@
...
-        ]
-    }
+				use: [
+					{
+						loader: "builtin:swc-loader",
+						options: {
+							sourceMap: true,
+							jsc: {
+								parser: {
+									syntax: "typescript",
```
Comment (thekip) [inline examples/rspack/rspack.config.js:30]: there is a problem with formatting. We use 2 spaces, here is used 4 spaces. Also part with "builtin:swc-loader" use even more spaces 

image

**Review by thekip — CHANGES_REQUESTED** [review]
I checked the example on my machine and it works with no issue. Thanks. 

I left one comment regarding code formatting (we have prettier switched off for `examples` folder, so you need to format it manually) 

Also, there is a paragraph in the Readme related to babel, it should be removed or changed to description that built-in loader is used in conjunction with lingui swc-plugin and you need to pick a right version of it.

File: examples/rspack/rspack.config.js (hunk 1 above)
Comment (aseerkt) [reply examples/rspack/rspack.config.js]: Thanks. I have addressed the changes you requested. PTAL.

**Review by thekip — APPROVED** [review]

**Review by andrii-bodnar — APPROVED** [review]
@aseerkt thank you!

**Pull request merged**